### PR TITLE
express-error-handling/

### PR DESCRIPTION
**Pull request by jeesupark96**

### CODE REVIEW

File: express-error-handling/server/index.js [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    .then(result => {
+      const grades = result.rows;
+      res.json(grades);
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```
Comment (scbowler) [inline express-error-handling/server/index.js]: In the catch callback we only need to call `next(err)` that will pass control to the error middleware, so we don't need to also send a response here.

```JAVASCRIPT
.catch(err => {
  next(err);
});
```

File: express-error-handling/server/index.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    .then(result => {
+      const [newGrade] = result.rows;
+      res.status(201).json(newGrade);
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```
Comment (scbowler) [inline express-error-handling/server/index.js]: Same here

File: express-error-handling/server/index.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```
Comment (scbowler) [inline express-error-handling/server/index.js]: Same here

File: express-error-handling/server/index.js [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      } else {
+        res.json(updatedGrade);
+      }
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```
Comment (scbowler) [inline express-error-handling/server/index.js]: Same here

File: express-error-handling/server/index.js [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      } else {
+        res.sendStatus(204);
+      }
+    })
+    .catch(err => {
+      next(err);
+      res.status(500).json({
+        error: 'an unexpected error occurred'
+      });
+    });
```
Comment (scbowler) [inline express-error-handling/server/index.js]: Same here

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, just needs updates to the catch callbacks. Let me know if you have any questions.

**Comment by jeesupark96** [comment]
got it!

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**